File: ArrowMove/src/test/GameTest.java
package test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import javax.swing.RootPaneContainer;

import game.controller.CommandLineController;
import game.data.Data;
import game.data.TextCommandData;
import game.rps.RPS;
import game.rps.data.RPSStateData;
import game.rps.shape.RPSResult;
import game.rps.shape.Shape;
import junit.framework.TestCase;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static game.rps.shape.RPSResult.*;
import static game.rps.shape.Shape.*;
@RunWith(JUnit4.class)
public class GameTest extends TestCase {
	RPS rps;
	CommandLineController clc;
	@Before
	public void setUp() throws Exception {

		rps = new RPS(null);
		clc = new CommandLineController();
	}

	@Test
	public void checkSHapeJudge() {
		assertEquals(ROCK.judge(ROCK),DRAW);
		assertEquals(ROCK.judge(PAPER),LOSE);
		assertEquals(ROCK.judge(SCISSOR),WIN);
		assertEquals(PAPER.judge(ROCK),WIN);
		assertEquals(PAPER.judge(PAPER),DRAW);
		assertEquals(PAPER.judge(SCISSOR),LOSE);
		assertEquals(SCISSOR.judge(ROCK),LOSE);
		assertEquals(SCISSOR.judge(PAPER),WIN);
		assertEquals(SCISSOR.judge(SCISSOR),DRAW);
		
	}
	@Test
	public void checkSHapeJudgeAll() {
		
		RPSResult[] results = judgeAll(SCISSOR,SCISSOR,SCISSOR,SCISSOR);
		//Arrays.stream(results).forEach(e->System.out.print(e+ ","));
		//System.out.println();
		
	}
	@Test
	public void checkText() {
		List<Shape>[] lists = new List[3];
		lists[0] = new ArrayList<Shape>(){{add(ROCK);add(PAPER);add(SCISSOR);}};
		lists[1] = new ArrayList<Shape>(){{add(ROCK);add(ROCK);add(ROCK);}};
		lists[2] = new ArrayList<Shape>(){{add(SCISSOR);add(PAPER);add(SCISSOR);}};

		RPSStateData rs = new RPSStateData(lists, 1);
		System.out.println(rs);
		
	}
	

}

File: ArrowMove/src/test/GameTest.kt
package test

import java.util.ArrayList
import java.util.Arrays

import javax.swing.RootPaneContainer

import game.controller.CommandLineController
import game.controller.group.ControllerGroup
import game.data.Data
import game.data.TextCommandData
import game.rps.RPS
import game.rps.controller.RPSRandom
import game.rps.data.RPSStateData
import game.rps.shape.RPSResult
import game.rps.shape.Shape

import org.junit.jupiter.api.*
import kotlin.test.assertEquals;

import game.rps.shape.RPSResult.*
import game.rps.shape.Shape.*
class GameTest{
    internal var rps: RPS = RPS(ControllerGroup( RPSRandom("test")));
    internal var clc: CommandLineController = CommandLineController()
    @BeforeEach
    @Throws(Exception::class)
    fun setUp() {

    }

    @Test
    fun checkSHapeJudge() {
        assertEquals(ROCK.judge(ROCK), DRAW)
        assertEquals(ROCK.judge(PAPER), LOSE)
        assertEquals(ROCK.judge(SCISSOR), WIN)
        assertEquals(PAPER.judge(ROCK), WIN)
        assertEquals(PAPER.judge(PAPER), DRAW)
        assertEquals(PAPER.judge(SCISSOR), LOSE)
        assertEquals(SCISSOR.judge(ROCK), LOSE)
        assertEquals(SCISSOR.judge(PAPER), WIN)
        assertEquals(SCISSOR.judge(SCISSOR), DRAW)

    }

    @Test
    fun checkSHapeJudgeAll() {

        val results = Shape.judgeAll(SCISSOR, SCISSOR, SCISSOR, SCISSOR)
        //Arrays.stream(results).forEach(e->System.out.print(e+ ","));
        //System.out.println();

    }

//    @Test
//    fun checkText() {
//        val lists = arrayOf<List<Shape>>(
//                object : ArrayList<Shape>() {
//            init {
//                add(ROCK)
//                add(PAPER)
//                add(SCISSOR)
//            }
//        }
//                , object : ArrayList<Shape>() {
//            init {
//                add(ROCK)
//                add(ROCK)
//                add(ROCK)
//            }
//        }
//                ,object : ArrayList<Shape>() {
//            init {
//                add(SCISSOR)
//                add(PAPER)
//                add(SCISSOR)
//            }
//        })
//
//    val rs = RPSStateData(lists, 1)
//    println(rs)
//
//}


}
